### PR TITLE
[core] feat(NumericInput, FileInput): add "small" modifier prop

### LINKED ISSUE #5035: Input - add small prop to numeric and file inputs
#### Feature request

Add `small` prop to [file inputs](https://blueprintjs.com/docs/#core/components/file-input) and [numeric inputs](https://blueprintjs.com/docs/#core/components/numeric-input).

[Tag inputs](https://blueprintjs.com/docs/#core/components/tag-input) also don't have a small version, but I don't think they should since tags have a minimum size...

#### Examples
[Text inputs](https://blueprintjs.com/docs/#core/components/text-inputs) have small versions.

Screen Shot 2021-11-24 at 12 31 46 AM

### PR DESCRIPTION
**Pull request by tmisirpash**
#### Checklist

- [ ] Includes tests
- [x] Update documentation

### CODE REVIEW

**Comment by palantirtech** [comment]
Thanks for your interest in [palantir/blueprint](https://github.com/palantir/blueprint), @tmisirpash! Before we can accept your pull request, you need to sign our contributor license agreement - just visit https://cla.palantir.com/ and follow the instructions. Once you sign, I'll automatically update this pull request.

**Review by adidahiya — CHANGES_REQUESTED** [review]
thanks @tmisirpash, can you please also add a demo for these new props to both `fileInputExample.tsx` and `numericInputBasicExample.tsx`?

File: packages/core/src/components/forms/_file-input.scss [hunk 1]
```diff
@@ -26,6 +27,10 @@ $file-input-button-width: $pt-grid-size * 7 !default;
 $file-input-button-width-large: $pt-grid-size * 8.5 !default;
 $file-input-button-padding: ($pt-input-height - $pt-button-height-small) * 0.5 !default;
 $file-input-button-padding-large: ($pt-input-height-large - $pt-button-height) * 0.5 !default;
+$file-input-button-width-small: $pt-grid-size * 5.5 !default;
+
+$pt-input-height-small-difference: $pt-input-height-small - $pt-button-height-smaller;
```
Comment (adidahiya) [inline packages/core/src/components/forms/_file-input.scss]: nit: no need for this variable, just inline its value on the next line

```suggestion
```

File: packages/core/src/components/forms/_file-input.scss [hunk 2]
```diff
@@ -26,6 +27,10 @@ $file-input-button-width: $pt-grid-size * 7 !default;
...
+$file-input-button-padding-small: $pt-input-height-small-difference * 0.5 !default;
```
Comment (adidahiya) [inline packages/core/src/components/forms/_file-input.scss]:
```suggestion
$file-input-button-padding-small: ($pt-input-height-small - $pt-button-height-smaller) * 0.5 !default;
```

**Review by adidahiya — CHANGES_REQUESTED** [review]
looks good in the [docs preview build](https://output.circle-artifacts.com/output/job/7bd24968-5878-4f65-8bcc-38fe9e2c2f15/artifacts/0/packages/docs-app/dist/index.html#core/components/file-input)! and thanks for adding CSS API support.

just left a few code style comments, otherwise this is good to go :shipit:

File: packages/core/src/components/forms/_file-input.scss (hunk 1 above)
Comment (tmisirpash) [reply packages/core/src/components/forms/_file-input.scss]: @adidahiya Done, but I had to disable the max-line-length lint rule for it to work

**Review by adidahiya — APPROVED** [review]

**Pull request merged**